pandas/tools/pivot.py

```diff
@@ -136,8 +136,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
```

Comment:
Hmm downstream of this block, `get_level_values` is only used if `margins` is True, which in this case is False.

Also, the [unstack docs](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.unstack.html) mentions that level names can be passed, so it appears that `unstack` functionally tries to figure out whether a position or name is passed.